typescript/infra/src/agents/index.ts

```diff
@@ -166,65 +147,72 @@ export async function getAgentEnvVars<Networks extends ChainName>(
         `OPT_VALIDATOR_VALIDATOR_TYPE=hexKey`,
       );
     }
-  } catch (error) {
-    // This happens if you don't have a result type
-    if ((error as any).toString().includes('Panic')) {
-      throw error;
+  } else {
+    // AWS keys
+
+    let user: AgentAwsUser<Networks>;
+
+    if (role === KEY_ROLE_ENUM.Validator) {
+      const checkpointSyncer =
+        agentConfig.validatorSets[outboxChainName].validators[index!]
+          .checkpointSyncer;
+      if (checkpointSyncer.type !== CheckpointSyncerType.S3) {
+        throw Error(
+          'Expected S3 checkpoint syncer for validator with AWS keys',
+        );
+      }
+      user = new ValidatorAgentAwsUser(
+        agentConfig.environment,
+        outboxChainName,
+        index!,
+        checkpointSyncer.region,
+        checkpointSyncer.bucket,
+      );
+    } else {
+      user = new AgentAwsUser(
+        agentConfig.environment,
+        outboxChainName,
+        role,
+        agentConfig.aws!.region,
+      );
     }
 
-    // Keys are in AWS
-    const awsKeys = await getSecretAwsCredentials(agentConfig);
+    const accessKeys = await user.getAccessKeys();
 
-    envVars.push(`AWS_ACCESS_KEY_ID=${awsKeys.accessKeyId}`);
-    envVars.push(`AWS_SECRET_ACCESS_KEY=${awsKeys.secretAccessKey}`);
+    envVars.push(`AWS_ACCESS_KEY_ID=${accessKeys.accessKeyId}`);
+    envVars.push(`AWS_SECRET_ACCESS_KEY=${accessKeys.secretAccessKey}`);
 
     // Only checkpointer and relayer need to sign txs
-    if (role === KEY_ROLE_ENUM.Checkpointer || role === KEY_ROLE_ENUM.Relayer) {
+    if (
+      role === KEY_ROLE_ENUM.Checkpointer ||
+      role === KEY_ROLE_ENUM.Relayer ||
+      role === KEY_ROLE_ENUM.Kathy
+    ) {
       chainNames.forEach((name) => {
-        const key = new AgentAwsKey(agentConfig, role, name);
-        envVars.push(`OPT_BASE_SIGNERS_${name.toUpperCase()}_TYPE=aws`);
-        envVars.push(
-          `OPT_BASE_SIGNERS_${name.toUpperCase()}_ID=${
-            key.credentialsAsHelmValue.aws.keyId
-          }`,
-        );
-        envVars.push(
-          `OPT_BASE_SIGNERS_${name.toUpperCase()}_REGION=${
-            key.credentialsAsHelmValue.aws.region
-          }`,
+        const key = new AgentAwsKey(agentConfig, name, role);
+        envVars = envVars.concat(
+          configEnvVars(key.keyConfig, 'BASE', 'SIGNERS_'),
         );
       });
     }
-
-    // Validator attestation key
-    if (role === KEY_ROLE_ENUM.Validator) {
-      const key = new AgentAwsKey(agentConfig, role, outboxChainName);
-      envVars.push(`OPT_BASE_VALIDATOR_TYPE=aws`);
-      envVars.push(
-        `OPT_BASE_VALIDATOR_ID=${key.credentialsAsHelmValue.aws.keyId}`,
-      );
-      envVars.push(
-        `OPT_BASE_VALIDATOR_REGION=${key.credentialsAsHelmValue.aws.region}`,
-      );
-    }
   }
 
   switch (role) {
     case KEY_ROLE_ENUM.Validator:
-      envVars.concat(
+      envVars = envVars.concat(
```

Comment:
You could just use push here?

Reply:
configEnvVars is an array, so pushing would result in envVars having an element that is an array